From 3b1e5f9585e5a1b8920387114c3831e957c85803 Mon Sep 17 00:00:00 2001
From: Sohail Ahmad <sohail@indionetworks.com>
Date: Tue, 4 Oct 2022 15:19:31 +0200
Subject: [PATCH 10/13] ipq40xx: Support for Indio's Outdoor IP67 WiFi5 AP
 UM-550AC

Changes made in OpenWrt supported targets to support Indio's UM-550AC AP which is based on IPQ4019 chipset

Signed-off-by: Sohail Ahmad <sohail@indionetworks.com>
---
 .../ipq40xx/base-files/etc/board.d/02_network      |  3 ++-
 .../etc/hotplug.d/firmware/11-ath10k-caldata       |  6 ++++--
 target/linux/ipq40xx/image/generic.mk              | 14 ++++++++++++++
 .../patches-5.10/901-arm-boot-add-dts-files.patch  |  3 ++-
 4 files changed, 22 insertions(+), 4 deletions(-)

diff --git a/target/linux/ipq40xx/base-files/etc/board.d/02_network b/target/linux/ipq40xx/base-files/etc/board.d/02_network
index 22409e32e7..b8a67f9ba7 100644
--- a/target/linux/ipq40xx/base-files/etc/board.d/02_network
+++ b/target/linux/ipq40xx/base-files/etc/board.d/02_network
@@ -153,7 +153,8 @@ ipq40xx_setup_interfaces()
 		ucidef_add_switch "switch0" \
 			"0u@eth0" "2:lan:4" "3:lan:3" "4:lan:2" "0u@eth1" "5:wan"
 		;;
-	um-325ac)
+	um-325ac |\
+	um-550ac) 
 		ucidef_set_interface_wan "eth0"
 		ucidef_set_interface_lan "eth1"
 		;;
diff --git a/target/linux/ipq40xx/base-files/etc/hotplug.d/firmware/11-ath10k-caldata b/target/linux/ipq40xx/base-files/etc/hotplug.d/firmware/11-ath10k-caldata
index 0e3977c075..715b19d565 100644
--- a/target/linux/ipq40xx/base-files/etc/hotplug.d/firmware/11-ath10k-caldata
+++ b/target/linux/ipq40xx/base-files/etc/hotplug.d/firmware/11-ath10k-caldata
@@ -142,7 +142,8 @@ case "$FIRMWARE" in
 		caldata_extract "0:ART" 4096 12064
 		ath10k_patch_mac $(mtd_get_mac_ascii 0:ART WLAN0_BASEMAC)
 		;;
-	um-325ac)
+	um-325ac |\
+	um-550ac) 
 		caldata_extract "ART" 0x1000 0x2f20
 		;;
 	esac
@@ -236,7 +237,8 @@ case "$FIRMWARE" in
 		caldata_extract "0:ART" 20480 12064
 		ath10k_patch_mac $(mtd_get_mac_ascii 0:ART WLAN1_BASEMAC)
 		;;
-	um-325ac)
+	um-325ac |\
+	um-550ac) 
 		caldata_extract "ART" 0x5000 0x2f20
 		;;
 	esac
diff --git a/target/linux/ipq40xx/image/generic.mk b/target/linux/ipq40xx/image/generic.mk
index dc67a8e5e1..59814e6792 100644
--- a/target/linux/ipq40xx/image/generic.mk
+++ b/target/linux/ipq40xx/image/generic.mk
@@ -1040,6 +1040,20 @@ define Device/unielec_u4019-32m
 endef
 TARGET_DEVICES += unielec_u4019-32m
 
+define Device/um-550ac
+	DEVICE_VENDOR := Indio Networks
+	DEVICE_MODEL := UM-550AC
+	BOARD_NAME := um-550ac
+	SOC := qcom-ipq4019
+	DEVICE_DTS := qcom-ipq4019-um-550ac
+	KERNEL_INSTALL := 1
+	KERNEL_SIZE := 4096k
+	IMAGE_SIZE := 26624k
+	$(call Device/FitImage)
+	IMAGE/sysupgrade.bin := append-kernel | pad-to $$$$(KERNEL_SIZE) | append-rootfs | pad-rootfs | append-metadata
+endef
+TARGET_DEVICES += um-550ac
+
 define Device/wallys_dr40x9
 	$(call Device/FitImage)
 	$(call Device/UbiFit)
diff --git a/target/linux/ipq40xx/patches-5.10/901-arm-boot-add-dts-files.patch b/target/linux/ipq40xx/patches-5.10/901-arm-boot-add-dts-files.patch
index b867469087..d66ffcaf1e 100644
--- a/target/linux/ipq40xx/patches-5.10/901-arm-boot-add-dts-files.patch
+++ b/target/linux/ipq40xx/patches-5.10/901-arm-boot-add-dts-files.patch
@@ -10,7 +10,7 @@ Signed-off-by: John Crispin <john@phrozen.org>
 
 --- a/arch/arm/boot/dts/Makefile
 +++ b/arch/arm/boot/dts/Makefile
-@@ -904,11 +904,79 @@ dtb-$(CONFIG_ARCH_QCOM) += \
+@@ -904,11 +904,80 @@ dtb-$(CONFIG_ARCH_QCOM) += \
  	qcom-apq8074-dragonboard.dtb \
  	qcom-apq8084-ifc6540.dtb \
  	qcom-apq8084-mtp.dtb \
@@ -76,6 +76,7 @@ Signed-off-by: John Crispin <john@phrozen.org>
 +	qcom-ipq4019-x1pro.dtb \
 +	qcom-ipq4019-u4019-32m.dtb \
 +	qcom-ipq4019-um-325ac.dtb \
++	qcom-ipq4019-um-550ac.dtb \
 +	qcom-ipq4019-wpj419.dtb \
 +	qcom-ipq4019-wtr-m2133hp.dtb \
 +	qcom-ipq4028-wpj428.dtb \
-- 
2.25.1

